contents/dotfiles/Dotfiles.md

```diff
@@ -0,0 +1,218 @@
+
+The example function `say_hello` below takes in 2 arguments, represented by `$1` and `$2`.
+```bash
+say_hello () {
+  echo "hello $1, good $2!"
+}
+```
+If the user were to type the following into the terminal:
+```bash
+say_hello John morning
+```
+it would produce the output:
+```
+hello John, good morning!
+```
+
+The use of functions essentially allows a user to create custom command-line tools to suit their needs.
+Defining the function above in `~/.bashrc` or `./.zshrc` will load it whenever a new shell is created.
+
+#### Benefit: Shell Utility and Styling
+
+Dotfiles are also often used to enhance or add features to a shell, such as `bash` or `zsh`.
+
+##### Feature: Prompt
+
+The `PS1` environment variable provides information for configuring and styling the Shell prompt. 
+By default, `PS1` is set to display only the username, host name and current working directory in the prompt.
+
+<pic src="default-prompt.png" alt="Default prompt" width="55%">
+
+  <sub> _Figure 2. Default prompt_ </sub>
+
+</pic>
+
+`PS1` can be [configured](https://www.cyberciti.biz/tips/howto-linux-unix-bash-shell-setup-prompt.html) to add (or remove) features, such as status code of the previous command, git branch, git status, dates, etc. In Figure 3 below, the prompt is able to detect and show the git branch and status when it is inside a git repository, a useful and convenient feature for developers.
+
+<pic src="enhanced-prompt.png" alt="Enhanced prompt" width="55%">
+
+  <sub> _Figure 3. Enhanced prompt_ </sub>
+
+</pic>
+
+##### Feature: Tab completion
+
+Tab completion is a feature that allows the user to select options from a drop down menu. This is a default feature when using `zsh`, but can be configured using dotfiles when using other shells like `bash`.
+
+<pic src="cd-autocomplete.gif" alt="Git flag autocomplete" width="60%">
+
+  <sub> _Figure 4. `cd` tab completion [(Tutorial)](https://scriptingosx.com/2019/07/moving-to-zsh-part-5-completions/)_ </sub>
+
+</pic>
+
+Tab completion can also be used for other purposes, such as selecting flags from a drop down menu, complete with an accompanying description.
+
+<pic src="git-flag-autocomplete.gif" alt="Git flag autocomplete" width="60%">
+
+  <sub> _Figure 5. Flag autocomplete_ </sub>
+
+</pic>
+
+##### Feature: Syntax Highlighting
+
+Dotfiles are also often used to style and colorise the terminal. This is not just for aesthetic reasons - the use of color enhances readability of programs and makes it easier to debug. One such example would be the use of **syntax highlighting** as seen in Figure 6 below. It is especially useful for long shell commands, escape sequences or string interpolation.
+
+<pic src="syntax-highlighting.gif" alt="Git flag autocomplete" width="60%">
+
+  <sub> _Figure 6 Syntax Highlighting [(Tutorial)](https://github.com/zsh-users/zsh-syntax-highlighting)_ </sub>
+
+</pic>
+
+### How to get Started with Dotfiles?
```

Comment:
Getting Started (consistent with other resources)

Reply:
This was based off the learning resource for Go